plinth/modules/dynamicdns/dynamicdns.py

```diff
@@ -0,0 +1,352 @@
+                             'timer': timer,
+                             'last_update': last_update,
+                             'subsubmenu': subsubmenu})
+
+
+def get_status():
+    """Return the current status"""
+    """ToDo: use key/value instead of hard coded value list"""
+    status = {}
+    output = actions.run('dynamicdns', 'status')
+    details = output.split()
+    status['enabled'] = (output.split()[0] == 'enabled')
+
+    if len(details) > 1:
+        if details[1] == 'disabled':
+            status['dynamicdns_server'] = ''
+        else:
+            status['dynamicdns_server'] = details[1]
+    else:
+        status['dynamicdns_server'] = ''
+
+    if len(details) > 2:
+        if details[2] == 'disabled':
+            status['dynamicdns_domain'] = ''
+        else:
+            status['dynamicdns_domain'] = details[2]
+    else:
+        status['dynamicdns_domain'] = ''
+
+    if len(details) > 3:
+        if details[3] == 'disabled':
+            status['dynamicdns_user'] = ''
+        else:
+            status['dynamicdns_user'] = details[3]
+    else:
+        status['dynamicdns_user'] = ''
+
+    if len(details) > 4:
+        if details[4] == 'disabled':
+            status['dynamicdns_secret'] = ''
+        else:
+            status['dynamicdns_secret'] = details[4]
+    else:
+        status['dynamicdns_secret'] = ''
+
+    if len(details) > 5:
+        if details[5] == 'disabled':
+            status['dynamicdns_ipurl'] = ''
+        else:
+            status['dynamicdns_ipurl'] = details[5]
+    else:
+        status['dynamicdns_ipurl'] = ''
+
+    if len(details) > 6:
+        if details[6] == 'disabled':
+            status['dynamicdns_update_url'] = ''
+        else:
+            status['dynamicdns_update_url'] = details[6]
+    else:
+        status['dynamicdns_update_url'] = ''
+
+    if len(details) > 7:
+        status['disable_SSL_cert_check'] = (output.split()[7] == 'enabled')
+    else:
+        status['disable_SSL_cert_check'] = False
+
+    if len(details) > 8:
+        status['use_http_basic_auth'] = (output.split()[8] == 'enabled')
+    else:
+        status['use_http_basic_auth'] = False
+
+    if not status['dynamicdns_server'] and not status['dynamicdns_update_url']:
+            status['service_type'] = '1'
+    elif not status['dynamicdns_server'] and status['dynamicdns_update_url']:
+            status['service_type'] = '4'
+    else:
+        status['service_type'] = '1'
+
+    return status
+
+
+def _apply_changes(request, old_status, new_status):
+    """Apply the changes to Dynamic DNS client"""
+    LOGGER.info('New status is - %s', new_status)
+    LOGGER.info('Old status was - %s', old_status)
+
+    if new_status['dynamicdns_secret'] == '':
+        new_status['dynamicdns_secret'] = old_status['dynamicdns_secret']
+
+    if new_status['dynamicdns_ipurl'] == '':
+        new_status['dynamicdns_ipurl'] = 'none'
+
+    if old_status['dynamicdns_server'] != \
```

Comment:
Could you also compare the dictionaries directly with `if old_status == new_status`?